### PR TITLE
GeneralWallet supports get_all_tx_list

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
GeneralWallet 支持 get_all_tx_list （先走 provider，非本地缓存）

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
https://github.com/OneKeyHQ/TaskHub/issues/1904

## Pull request type
 
_Put an `x` in the boxes that apply_
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

## Any other comments?
…

### CODE REVIEW

File: electrum_gui/android/console.py [hunk 1]
```diff
@@ -1686,25 +1686,25 @@ def get_btc_tx_list(self, start=None, end=None, search_type=None):  # noqa
...
-    def get_eth_tx_list(self, contract_address=None, search_type=None):
+    def get_general_coin_tx_list(self, contract_address=None, search_type=None):
         ret = []
 
         chain_code = coin_manager.legacy_coin_to_chain_code(self.wallet.coin)
-        main_coin = coin_manager.get_coin_info(coin_manager.get_chain_info(chain_code).fee_code)
+        main_coin = coin_manager.get_coin_info(coin_manager.get_chain_info(chain_code).chain_code)
         main_coin_price = price_manager.get_last_price(main_coin.code, self.ccy)
         if contract_address is not None:
-            contract_address = contract_address.lower()
```
Comment (taimanhui) [inline electrum_gui/android/console.py:1696]: 数据库的部分是没问题，但是COIN_DICT的部分并没忽略大小写（不过COIN_DICT部分应该没有token，以后应该也没有才对）。所以这里留着防也行，直接删了也无所谓

**Review by taimanhui — APPROVED** [review]
你这几个commit都比较简单其实可以合并在一个PR避免自己还要多次rebase的 😂

File: electrum_gui/android/console.py (hunk 1 above)
Comment (huazhouwang) [reply electrum_gui/android/console.py:1696]: COIN_DICT 这部分还好，相对可控，DB 查询会忽略大小写。因为其他链，例如 solana 是大小写敏感的，全小写了反而有问题

**Comment by huazhouwang** [comment]
并入 https://github.com/OneKeyHQ/electrum/pull/798

**Pull request closed without merging**